modules/aruco/src/aruco.cpp

```diff
@@ -1542,22 +1542,22 @@ void _drawPlanarBoardImpl(Board *_board, Size outSize, OutputArray _img, int mar
             // remove negativity
             p1.x = p0.x - minX;
             p1.y = p0.y - minY;
-            pf.x = p1.x * float(markerZone.cols - 1) / sizeX;
-            pf.y = float(markerZone.rows - 1) - p1.y * float(markerZone.rows - 1) / sizeY;
+            pf.x = p1.x * float(markerZone.cols) / sizeX;
```

Comment:
according to [this thread](http://answers.opencv.org/question/87923/sub-pixel-coordinate-origin/) the location `(0,0)` refers to the center of the top left pixel.

This would then mean we have to write
`pf.x = p1.x * float(markerZone.cols - 1) / sizeX - 0.5;`
still need to verify this however..